exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/project/ProjectRecordBatch.java

```diff
@@ -339,6 +331,14 @@ protected IterOutcome handleNullInput() {
     return IterOutcome.OK_NEW_SCHEMA;
   }
 
+  private boolean isNotSchemalessInput() {
+    RecordBatch incomingBatch = incoming instanceof IteratorValidatorBatchIterator
+        ? ((IteratorValidatorBatchIterator) incoming).getIncoming()
+        : incoming;
```

Comment:
Is it required to ensure that incoming batch after `IteratorValidatorBatchIterator` is `IteratorValidatorBatchIterator`?

Looks like `SchemalessBatch.getSchema()` always returns null, so at least the second check covers this `instance of` check...